File: py/test/data_structure_tests/StackTest.py
import unittest
from py.data_structures.Stack import Stack

class tests(unittest.TestCase):
    def test1(self):
        new_stack = Stack()
        self.assertRaises(IndexError, new_stack.pop)
        self.assertRaises(IndexError, new_stack.top)

    def test2(self):
        new_stack = Stack()
        for i in range(150):
            new_stack.push(i)
        
        for i in reversed(range(150)):
            self.assertEqual(len(new_stack), i+1)
            self.assertEqual(new_stack.top(), i)
            self.assertEqual(new_stack.pop(), i) 

if __name__ == "__main__":
    unittest.main() 

File: ts/src/tests/data_structure_tests/stack.test.ts
import Stack from '../../data structures/stack';

test('test1', () => {
    const newStack = new Stack();

    expect(() => newStack.top()).toThrow('empty stack');
    expect(() => newStack.pop()).toThrow('pop from empty stack');
});

test('test2', () => {
    const newStack = new Stack();
    for (let i = 0; i < 150; i += 1) {
        newStack.push(i);
    }

    for (let i = 149; i >= 0; i -= 1) {
        expect(newStack.length()).toEqual(i + 1);
        expect(newStack.top()).toEqual(i);
        expect(newStack.pop()).toEqual(i);
    }
});
